@memZono/memZono.m

```diff
@@ -376,9 +411,33 @@
         obj = combine(obj1,obj2); % Minkowski Sum
 
         % Additional Methods
-        function out = linMap(in,M,outDims)
+        function out = linMap(in,M,inDims,outDims)
+            % if isscalar(varargin)
+            %     inDims = in.dimKeys;
+            %     outDims = varargin{1}; 
+            %     warning('lack of inDims specification can cause issues with dimension ordering'); 
+            % else
+            %     inDims = varargin{1};
+            %     outDims = varargin{2};
+            % end
+            if ~iscell(inDims); inDims = memZono.genKeys(inDims,1:size(M,1)); end
             if ~iscell(outDims); outDims = memZono.genKeys(outDims,1:size(M,1)); end
-            out = in.transform([],M,[],outDims);
+            out = in.transform([],M,inDims,outDims).projection(outDims);
```

Comment:
I'm not sure we need the projection after the call to transform....?

Reply:
I'm not sure if it should be needed (that's a further disscussion) but as it is now the non-specified dimensions are just passed through the transformation, thus a projection is required to only output the appropriete dimensions for linMap() if the original set has extra dimensions that aren't needed for output